src/pages/Loans/LoansOverview/components/LoansMarkets/SupplyMarket.tsx

```diff
@@ -0,0 +1,131 @@
+import { CurrencyIdLiteral } from '@interlay/interbtc-api';
+import { Key, useState } from 'react';
+
+import { ReactComponent as ChevronRight } from '@/assets/img/icons/chevron-right.svg';
+import { CoinIcon, Span, Stack } from '@/component-library';
+import { SupplyAssetData, SupplyPositionData } from '@/utils/hooks/api/loans/use-get-loans-data';
+
+import { SupplyModal } from '../SupplyModal';
+import { StyledApyTag, StyledAsset, StyledTableWrapper } from './LoansMarkets.style';
+import { MarketTable } from './MarketTable';
+import { SupplyAssetsColumns, SupplyAssetsTableRow, SupplyPositionColumns, SupplyPositionTableRow } from './types';
+
+// TODO: translations
+const supplyAssetsColumns = [
+  { name: 'Asset', uid: SupplyAssetsColumns.ASSET },
+  { name: 'APY', uid: SupplyAssetsColumns.APY },
+  { name: 'Wallet Balance', uid: SupplyAssetsColumns.WALLET_BALANCE },
+  { name: '', uid: SupplyAssetsColumns.END_ADORNMENT }
+];
+
+// TODO: translations
+const supplyPositionColumns = [
+  { name: 'Asset', uid: SupplyPositionColumns.ASSET },
+  { name: 'Supplied', uid: SupplyPositionColumns.SUPPLIED },
+  { name: 'Supply APY', uid: SupplyPositionColumns.SUPPLY_APY },
+  { name: 'APY Earned', uid: SupplyPositionColumns.SUPPLY_APY },
+  { name: '', uid: SupplyPositionColumns.END_ADORNMENT }
+];
+
+type UseAssetState = {
+  data?: SupplyAssetData;
+  position?: SupplyPositionData;
+};
+
+const defaultAssetState: UseAssetState = { data: undefined, position: undefined };
+
+type SupplyMarketProps = {
+  assets: SupplyAssetData[];
+  positions: SupplyPositionData[];
+};
+
+const MarketAsset = ({ currency }: { currency: CurrencyIdLiteral }): JSX.Element => (
+  <StyledAsset>
+    <CoinIcon coin={currency} size='small' />
+    <Span>{currency}</Span>
+  </StyledAsset>
+);
+
+const SupplyMarket = ({ assets, positions }: SupplyMarketProps): JSX.Element => {
+  const [selectedAsset, setAsset] = useState<UseAssetState>(defaultAssetState);
+
+  const handleAssetRowAction = (key: Key) => {
```

Comment:
quite a bit of duplication here - can `handleAssetRowAction` and `handlePositionRowAction` be consolidated?

Reply:
Really unsure if this is going to be kept in the future. It did it this way because of my current data structure. @peterslany is working on the data points we need, so this is very much subject to change.